core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -577,6 +575,7 @@ public static void deleteCancerStudy(int internalCancerStudyId) throws DaoExcept
         ResultSet rs = null;
         try {
             con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            deleteGenericAssayMeta(internalCancerStudyId);
```

Comment:
I checked the database scheme, `genetic_alteration` has a constrain **FOREIGN KEY (`GENETIC_ENTITY_ID`) REFERENCES `genetic_entity` (`ID`) ON DELETE CASCADE**. I think generic assay related data can be deleted before we delete the data in `genetic_alteration` table.

Reply:
I was thinking the check should be wrapped around the call here to prevent the method call (and grabbing of db connection - https://github.com/cBioPortal/cbioportal/blob/9c6c3f7d69ede2f7c5e8ee9bbea626dc833b10a7/core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java#L578